tests/extended_temporal_memory/extensive_etm_test.py

```diff
@@ -0,0 +1,1340 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import unittest
+import random
+
+from nupic.data.generators.pattern_machine import PatternMachine
+from nupic.support.unittesthelpers.abstract_temporal_memory_test import AbstractTemporalMemoryTest
+
+from htmresearch.algorithms.extended_temporal_memory import ExtendedTemporalMemory
+
+
+class ExtensiveExtendedTemporalMemoryTest(AbstractTemporalMemoryTest,
+                                          unittest.TestCase):
+  """
+  Tests the specific aspects of extended temporal memory (external and apical
+  input, learning on one cell, etc.
+
+  The parameters used in those tests are the typical default parameters for
+  temporal memory, unless stated otherwise in the experiment (when self.init()
+  is called).
+
+      columnDimensions: (2048,)
+      cellsPerColumn: 32
+      initialPermanence: 0.5
+      connectedPermanence: 0.6
+      minThreshold: 25
+      maxNewSynapseCount: 30
+      permanenceIncrement: 0.1
+      permanenceDecrement: 0.02
+      predictedSegmentDecrement: 0.08
```

Comment:
Typically we use a much smaller value for predictedSegmentDecrement, about 10% or 5% of the increment